src/analysis/processing/qgsalgorithmlinedensity.cpp

```diff
@@ -0,0 +1,193 @@
+  QFileInfo fi( outputFile );
+  const QString outputFormat = QgsRasterFileWriter::driverForExtension( fi.suffix() );
+
+  int rows = std::max( std::ceil( mExtent.height() / mPixelSize ), 1.0 );
+  int cols = std::max( std::ceil( mExtent.width() / mPixelSize ), 1.0 );
+
+  //build new raster extent based on number of columns and cellsize
+  //this prevents output cellsize being calculated too small
+  QgsRectangle rasterExtent = QgsRectangle( mExtent.xMinimum(), mExtent.yMaximum() - ( rows * mPixelSize ), mExtent.xMinimum() + ( cols * mPixelSize ), mExtent.yMaximum() );
+
+  std::unique_ptr< QgsRasterFileWriter > writer = qgis::make_unique< QgsRasterFileWriter >( outputFile );
+  writer->setOutputProviderKey( QStringLiteral( "gdal" ) );
+  writer->setOutputFormat( outputFormat );
+  std::unique_ptr<QgsRasterDataProvider > provider( writer->createOneBandRaster( Qgis::Float32, cols, rows, rasterExtent, mCrs ) );
+  if ( !provider )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output: %1" ).arg( outputFile ) );
+  if ( !provider->isValid() )
+    throw QgsProcessingException( QObject::tr( "Could not create raster output %1: %2" ).arg( outputFile, provider->error().message( QgsErrorMessage::Text ) ) );
+
+  provider->setNoDataValue( 1, -9999 );
+
+  int totalCellcnt = rows * cols;
+  int cellcnt = 0;
+
+  for ( int row = 0; row < rows; row++ )
+  {
+    std::unique_ptr< QgsRasterBlock > rasterDataLine = qgis::make_unique< QgsRasterBlock >( Qgis::Float32, cols, 1 );
+    for ( int col = 0; col < cols; col++ )
+    {
+      if ( feedback->isCanceled() )
+      {
+        break;
+      }
+
+      if ( col > 0 )
+        mSearchGeometry.translate( mPixelSize, 0 );
+
+      QList<QgsFeatureId> fids = mIndex.intersects( mSearchGeometry.boundingBox() );
+
+      std::unique_ptr< QgsGeometryEngine > engine( QgsGeometry::createGeometryEngine( mSearchGeometry.constGet() ) );
+      engine->prepareGeometry();
+
+      //remove non-intersecting fids
+      int i = 0;
+      while ( i < fids.size() )
+      {
+        QgsFeatureId fid = fids.at( i );
+        QgsGeometry lineGeom = mIndex.geometry( fid );
+
+        if ( engine->intersects( lineGeom.constGet() ) == false )
+        {
+          fids.removeAt( i );
+        }
+        i++;
+      }
+
+      QgsFeatureIds isectingFids = QSet< QgsFeatureId >::fromList( fids );
+      QgsFeatureIterator fit = mSource->getFeatures( QgsFeatureRequest().setFilterFids( isectingFids ) );
+      QgsFeature f;
+      double abs_density = 0;
+      while ( fit.nextFeature( f ) )
+      {
+        double analysisLineLength =  mDa.measureLength( QgsGeometry( engine->intersection( f.geometry().constGet() ) ) );
+        double analysisWeight = f.attribute( mWeight ).toDouble();
+        abs_density += ( analysisLineLength * analysisWeight );
+      }
+
+      double lineDensity = abs_density / mSearchGeometryArea;
```

Comment:
To be exact -- you should be calculating the search geometry area here, and not using the earlier version. You've used an ellipsoid based calculation for the line length, and potentially the area of the buffer circle will change based on the lat/long of the actual pixel being calculated.